include/mbedtls/ecdh.h

```diff
@@ -40,168 +51,210 @@ typedef enum
 } mbedtls_ecdh_side;
 
 /**
- * \brief           ECDH context structure
+ * \brief           The ECDH context structure.
  */
 typedef struct
 {
-    mbedtls_ecp_group grp;      /*!<  elliptic curve used                           */
-    mbedtls_mpi d;              /*!<  our secret value (private key)                */
-    mbedtls_ecp_point Q;        /*!<  our public value (public key)                 */
-    mbedtls_ecp_point Qp;       /*!<  peer's public value (public key)              */
-    mbedtls_mpi z;              /*!<  shared secret                                 */
-    int point_format;   /*!<  format for point export in TLS messages       */
-    mbedtls_ecp_point Vi;       /*!<  blinding value (for later)                    */
-    mbedtls_ecp_point Vf;       /*!<  un-blinding value (for later)                 */
-    mbedtls_mpi _d;             /*!<  previous d (for later)                        */
+    mbedtls_ecp_group grp;   /*!< The elliptic curve used. */
+    mbedtls_mpi d;           /*!< The secret value of our private key. */
+    mbedtls_ecp_point Q;     /*!< The public value of our public key. */
+    mbedtls_ecp_point Qp;    /*!< The value of the public key of the peer. */
+    mbedtls_mpi z;           /*!< The shared secret. */
+    int point_format;        /*!< The format of point export in TLS messages. */
+    mbedtls_ecp_point Vi;    /*!< The blinding value. */
+    mbedtls_ecp_point Vf;    /*!< The unblinding value. */
+    mbedtls_mpi _d;          /*!< The previous \p d. */
 }
 mbedtls_ecdh_context;
 
 /**
- * \brief           Generate a public key.
- *                  Raw function that only does the core computation.
+ * \brief           This function generates a public key.
+ *
+ *                  This raw function only performs the core computation.
```

Comment:
I think the pre-existing wording is not adequate: This function generates a public/private keypair, and the function documentation puts more emphasis is on the public part of the key for no reason. Suggestion: `Generate an ECDH keypair on an elliptic curve.`

Reply:
Fixed

Reply:
I'd also suggest to drop the second sentence altogether, because the use of `only` for one of the two core functions of ECDH feels inappropriate. As an alternative, one might use `This function performs the first of the two core computations during the ECDH key exchange, the second one being implemented by mbedtls_ecdh_compute_shared.`

Reply:
Fixed